components/brave_wallet_ui/page/screens/swap/hooks/useSwap.ts

```diff
@@ -848,8 +852,39 @@ export const useSwap = () => {
     return Amount.zero()
   }, [quoteOptions])
 
+  const availableProvidersForSwap = useMemo(() => {
+    // Auto and Li.Fi are always supported.
+    const defaultProviders = [
+      BraveWallet.SwapProvider.kAuto,
+      BraveWallet.SwapProvider.kLiFi
+    ]
+    // Bridge only supportes Auto and Li.Fi.
+    if (isBridge) {
+      return defaultProviders
+    }
+    // Returns all providers if no from token has been selected yet.
+    if (!fromToken) {
+      return [
+        ...defaultProviders,
+        BraveWallet.SwapProvider.kZeroEx,
+        BraveWallet.SwapProvider.kJupiter
+      ]
```

Comment:
Perf: can be defined outside of the hook

Reply:
Moved!